napari/components/viewer_model.py

```diff
@@ -706,16 +724,23 @@ def open_sample(
             data = _sample_data[plugin][sample]['data']
         except KeyError:
             samples = available_samples()
-            msg = trans._(
-                f"Plugin {plugin!r} does not provide sample data "
-                f"named {sample!r}. "
-            )
             if samples:
-                msg += trans._(
-                    'Available samples include: {samples}', samples=samples
+                msg = trans._(
+                    "Plugin {plugin} does not provide sample data named {sample}. Available samples include: {samples}.",
+                    deferred=True,
+                    plugin=f"{plugin!r}",
+                    sample=f"{sample!r}",
+                    samples=samples,
                 )
             else:
-                msg += trans._("No plugin samples have been registered.")
+                msg = trans._(
+                    "Plugin {plugin} does not provide sample data named {sample}. No plugin samples have been registered.",
+                    deferred=True,
+                    plugin=f"{plugin!r}",
+                    sample=f"{sample!r}",
+                    samples=samples,
+                )
```

Comment:
why not
```suggestion
                msg = trans._(
                    "Plugin {plugin!r} does not provide sample data named {sample!r}. No plugin samples have been registered.",
                    deferred=True,
                    plugin=plugin,
                    sample=sample,
                    samples=samples,
                )
```
?

And `samples` is not used.

Reply:
note: I get why we don't use expression like in f-strings; but I thoughts that formatter like `!r`, `%02d` and similar would work.

Reply:
Yes agreed!, will revert all of these!

Reply:
Fixed